x-pack/plugins/observability/public/components/custom_threshold/components/alert_details_app_section/alert_history.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import React from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { RuleTypeParams } from '@kbn/alerting-plugin/common';
+import { EventAnnotationConfig } from '@kbn/event-annotation-common';
+import { DataView } from '@kbn/data-views-plugin/common';
+import {
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiText,
+  EuiSpacer,
+  EuiLoadingSpinner,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ALERT_GROUP, ALERT_GROUP_VALUE, type AlertConsumers } from '@kbn/rule-data-utils';
+import { useAlertsHistory } from '@kbn/observability-alert-details';
+import { convertTo } from '../../../../../common/utils/formatters';
+import { useKibana } from '../../../../utils/kibana_react';
+import { AlertParams } from '../../types';
+import { getGroupFilters, getGroupQueries } from '../helpers/get_group';
+import { RuleConditionChart } from '../rule_condition_chart/rule_condition_chart';
+import { CustomThresholdAlert, CustomThresholdRule } from '../types';
+
+const DEFAULT_INTERVAL = '1d';
+const SERIES_TYPE = 'bar_stacked';
+
+interface Props {
+  alert: CustomThresholdAlert;
+  rule: CustomThresholdRule;
+  dataView?: DataView;
+}
+
+const dateRange = {
+  from: 'now-30d',
+  to: 'now',
+};
+
+export function AlertHistoryChart({ rule, dataView, alert }: Props) {
+  const { http, notifications } = useKibana().services;
+  const { euiTheme } = useEuiTheme();
+  const ruleParams = rule.params as RuleTypeParams & AlertParams;
+  const criterion = rule.params.criteria[0];
+  const groups = alert.fields[ALERT_GROUP];
+  const featureIds = [rule.consumer as AlertConsumers];
+
+  const {
+    data: { histogramTriggeredAlerts, avgTimeToRecoverUS, totalTriggeredAlerts },
+    isLoading,
+    isError,
+  } = useAlertsHistory({
+    http,
+    featureIds,
+    ruleId: rule.id,
+    dateRange,
+    queries: getGroupQueries(groups, ALERT_GROUP_VALUE),
```

Comment:
@maryam-saeidi In which scenario, we would not use `kibana.alert.group.value` as term field in `getGroupQueries` function, why do we need to pass `fieldName` (ALERT_GROUP_VALUE) separately here?

Reply:
`useAlertsHistory` queries the AAD indices instead of the indices that are used during alert execution (based on the selected data view), so I needed to adjust the field that we use to filter based on group information.

Reply:
hmmm, couldn't we directly use `ALERT_GROUP_VALUE` [here](https://github.com/elastic/kibana/blob/2aaced567922bb4f0899dcbf752de7d709e28ee6/x-pack/plugins/observability/public/components/custom_threshold/components/helpers/get_group.ts#L17) instead of `[fieldName || group.field]`?

Reply:
No, because in other use cases thay we use this function, we don't pass the fieldName and as a result we use the field from the group itself.

Reply:
Ah, I see why it is confusing, in other cases I am using `getGroupFilters` without fieldName but behind the scene, it uses `getGroupQueries`.

Reply:
I think it would be helpful to change `fieldName` to something like `alertGroupField` and add a comment about its usage.

Reply:
I like the idea of adding a comment, I'll do that.
About the name, what do you think about `groupFieldName` if that is more clear? I don't want to limit it to alert in case we want to use this logic for other use cases (Although I don't have any example of such a use case in mind for now.)

Reply:
`groupFieldName` sounds good too.